### PR TITLE
PR: Migrate History Plugin to new API

### LINKED ISSUE #12190: Move History plugin to use new API
- [x] Move to new API
- [x] Ensure tests pass

### PR DESCRIPTION
**Pull request by goanpeca**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #12190

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

**Comment by goanpeca** [comment]
Rebased and ready for review @ccordoba12

**Comment by dalthviz** [comment]
Seems like some tests are failing could you check @goanpeca ?

**Comment by goanpeca** [comment]
Will do!

**Comment by goanpeca** [comment]
Fixed @dalthviz !

**Comment by dalthviz** [comment]
Seems like a rebase/merge is needed after the Help pane PR was merged and some tests are failing, could you take a look @goanpeca ? Thanks!

**Comment by goanpeca** [comment]
Looking into it @dalthviz !

**Comment by goanpeca** [comment]
>Seems like a rebase/merge is needed after the Help pane PR was merged

Fixed :-) 

>and some tests are failing,

It seems unrelated to this PR since the failure was a Comm timeout error and only on pip slow.

Thanks @dalthviz

**Comment by goanpeca** [comment]
>A traceback appears in the anaconda prompt (maybe this was caused since the dev pyls version needed to be installed in the env I'm using?):

Does is a known issue being fixed by https://github.com/spyder-ide/spyder/pull/12965

>While checking I realized that you can add breakpoints to the history (same behavior is present in Spyder 4 but could be worthy to check in a different issue if that's something intended or not):

Hmm yeah, will disable that, should be easy.

>Seems like changing the color scheme doesn't affect the plugin:

Easy fix.

Thanks for the QA work @dalthviz!

**Comment by goanpeca** [comment]
Hi @dalthviz, I think I solved the issues you reported. Thanks!

**Comment by goanpeca** [comment]
This one is ready for review @ccordoba12

File: spyder/plugins/history/plugin.py [hunk 1]
```diff
@@ -4,261 +4,97 @@
...
+    CONF_WIDGET_CLASS = HistoryConfigPage
+    CONF_FILE = False
+    CONF_FROM_OPTIONS = {
+        'color_scheme_name': ('appearance', 'selected'),
+    }
+    # --- Signals
+    # ------------------------------------------------------------------------
+    sig_focus_changed = Signal()
+    """
+    This signal is emitted when the focus of the code editor storing the
```
Comment (ccordoba12) [inline spyder/plugins/history/plugin.py:43]: storing ~~the~~

File: spyder/plugins/history/widgets.py [hunk 2]
```diff
@@ -8,155 +8,352 @@
...
+        'go_to_eof': True,
+        'line_numbers': True,
+        'max_entries': 100,
+        'wrap': True,
+    }
+
+    # Signals
+    sig_focus_changed = Signal()
+    """
+    This signal is emitted when the focus of the code editor storing the
```
Comment (ccordoba12) [inline spyder/plugins/history/widgets.py:67]: storing ~~the~~

Comment (goanpeca) [reply spyder/plugins/history/widgets.py]: Fixed

File: spyder/plugins/history/plugin.py (hunk 1 above)
Comment (goanpeca) [reply spyder/plugins/history/plugin.py]: Fixed

**Comment by ccordoba12** [comment]
This one has a merge conflict now.

**Comment by goanpeca** [comment]
Fixed conflicts.

**Comment by ccordoba12** [comment]
@goanpeca, please skip this test on macOS:

    spyder/plugins/onlinehelp/tests/test_pydocgui.py::test_pydocbrowser

because it's failing recurrently on that platform.

**Pull request merged**